client/session.go

```diff
@@ -1774,12 +1780,20 @@ func (s *session) streamBlocksMetadataFromPeerV2(
 
 		for _, elem := range result.Elements {
 			blockStart := time.Unix(0, elem.Start)
+
+			// Reset the reused ID data
+			reusedIDBytes.Resize(0)
+			reusedIDBytes.AppendAll(elem.ID)
+
+			// Clone the ID
+			clonedID := idPool.Clone(reusedID)
```

Comment:
For some unrelated changes, I'm adding some methods to the ident.Pool to make this easier. Do me a solid and drop a `TODO(prateek): undo reusedIDBytes .. once https://github.com/m3db/m3x/pull/125 is merged` here